pkg/scheduler/reconciliation/filter.go

```diff
@@ -8,12 +8,106 @@ import (
 	"github.com/kyma-incubator/reconciler/pkg/model"
 )
 
+type FilterMixer struct {
+	Filters []Filter
+}
+
+func (fm *FilterMixer) FilterByQuery(q *db.Select) error {
+	for i := range fm.Filters {
+		if err := fm.Filters[i].FilterByQuery(q); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (fm *FilterMixer) FilterByInstance(re *model.ReconciliationEntity) *model.ReconciliationEntity {
+	entity := re
+	for i := range fm.Filters {
+		if entity == nil {
+			break
+		}
+		entity = fm.Filters[i].FilterByInstance(entity)
+	}
+	return entity
+}
+
+type Limit struct {
+	Count       int
+	actualCount int
+}
+
+func (l *Limit) FilterByQuery(q *db.Select) error {
+	q.Limit(l.Count)
+	return nil
+}
+
+func (l *Limit) FilterByInstance(re *model.ReconciliationEntity) *model.ReconciliationEntity {
+	if l.actualCount < l.Count {
+		l.actualCount++
+		return re
+	}
+	return nil
+}
+
+type WithStatuses struct {
+	Statuses []string
+}
+
+func (ws *WithStatuses) FilterByQuery(q *db.Select) error {
+	if len(ws.Statuses) < 1 {
+		return nil
+	}
+
+	var whereRaw string
+	for i := range ws.Statuses {
+		if i > 0 {
+			whereRaw = fmt.Sprintf("%s%s", whereRaw, " OR ")
+		}
+
+		whereRaw = fmt.Sprintf("%sstatus='%s'", whereRaw, ws.Statuses[i])
```

Comment:
Please use db.ColumnHandler for retrieving column-names, otherwise we wont notice DB-schema discrepancies properly:

		statusColHandler, err := db.NewColumnHandler(&model.ReconciliationEntity{}, q.Conn, q.Logger)
		if err != nil {
			return err
		}
		statusColName, err := statusColHandler.ColumnName("Status")

JFYI: the `conn` field in the Query object can be made public, currently it is private.

Reply:
Thanks for this hint!